packages/cosmic-swingset/test/unitTests/test-lib-wallet.js

```diff
@@ -0,0 +1,267 @@
+      moolaBundle.amountMath.make(100),
+      `deposit successful`,
+    );
+    t.deepEquals(
+      wallet.getIssuerNames(moolaBundle.issuer),
+      {
+        issuerPetname: 'moola',
+        brandRegKey: 'fakeRegKeyMoola',
+      },
+      `returns petname and brandRegKey`,
+    );
+    t.deepEquals(pursesStateChangeLog, [
+      '[{"issuerPetname":"moola","brandRegKey":"fakeRegKeyMoola","pursePetname":"fun money","extent":0}]',
+    ]);
+    t.deepEquals(inboxStateChangeLog, []);
+  } catch (e) {
+    t.isNot(e, e, 'unexpected exception');
+  } finally {
+    t.end();
+  }
+});
+
+test('lib-wallet offer methods', async t => {
+  try {
+    const {
+      moolaBundle,
+      wallet,
+      instanceRegKey,
+      registry,
+      inboxStateChangeLog,
+      pursesStateChangeLog,
+    } = await setupTest();
+
+    const moolaBrandRegKey = registry.register('moolaBrand', moolaBundle.brand);
+    await wallet.addIssuer('moola', moolaBundle.issuer, moolaBrandRegKey);
+    await wallet.makeEmptyPurse('moola', 'Fun budget');
+    await wallet.deposit(
+      'Fun budget',
+      moolaBundle.mint.mintPayment(moolaBundle.amountMath.make(100)),
+    );
+    const formulateBasicOffer = id =>
+      harden({
+        // JSONable ID for this offer.  This is scoped to the origin.
+        id,
+
+        // Contract-specific metadata.
+        instanceRegKey,
+
+        // Format is:
+        //   hooks[targetName][hookName] = [hookMethod, ...hookArgs].
+        // Then is called within the wallet as:
+        //   E(target)[hookMethod](...hookArgs)
+        hooks: {
+          publicAPI: {
+            getInvite: ['makeInvite'], // E(publicAPI).makeInvite()
+          },
+        },
+
+        proposalTemplate: {
+          give: {
+            Contribution: {
+              // The pursePetname identifies which purse we want to use
+              pursePetname: 'Fun budget',
+              extent: 1,
+            },
+          },
+          exit: { onDemand: null },
+        },
+      });
+
+    const rawId = '1588645041696';
+    const id = `unknown#${rawId}`;
+    const offer = formulateBasicOffer(rawId);
+
+    const hooks = wallet.hydrateHooks(offer.hooks);
+    await wallet.addOffer(offer, hooks);
+
+    t.deepEquals(
+      wallet.getOffers(),
+      [
+        {
+          id,
+          instanceRegKey: 'automaticrefundinstancehandle_3467',
+          hooks: { publicAPI: { getInvite: ['makeInvite'] } },
+          proposalTemplate: {
+            give: {
+              Contribution: {
+                pursePetname: 'Fun budget',
+                extent: 1,
+                issuerPetname: 'moola',
+                brandRegKey: 'moolabrand_2059',
+              },
+            },
+            exit: { onDemand: null },
+          },
+          requestContext: { origin: 'unknown' },
+          status: undefined,
+        },
+      ],
+      `offer structure`,
+    );
+    const { outcome } = await wallet.acceptOffer(id);
```

Comment:
The wallet used to return `depositedP` which allowed you to synchronise with the actual deposit completion.  It seems that functionality has been lost.  There is no way to know if the spoils have been deposited, so I think your test wins the race before that happens.

Reply:
Gotcha, I must have removed that. I'll add it back. thanks!

Reply:
That's added back, and I did a manual test with dapp-encouragement as well